### PR TITLE
Support to configure init container resource for TiFlash

### LINKED ISSUE #4303: Resource requirements missed in the init container
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

Refer to https://asktug.com/t/topic/274069
**What did you expect to see?**

**What did you see instead?**

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

Close #4303
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [x] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support to configure init container resource for TiFlash
```

### CODE REVIEW

File: pkg/manager/member/utils.go [hunk 1]
```diff
@@ -266,6 +266,15 @@ func MapContainers(podSpec *corev1.PodSpec) map[string]corev1.Container {
 	return m
 }
 
+// MapContainers index init containers of Pod by container name in favor of looking up
+func MapInitContainers(podSpec *corev1.PodSpec) map[string]corev1.Container {
```
Comment (csuzhangxc) [inline pkg/manager/member/utils.go:270]: where do we use this function?

Comment (KanShiori) [reply pkg/manager/member/utils.go:270]: now use it in ut

**Review by csuzhangxc — APPROVED** [review]

File: pkg/manager/member/tiflash_member_manager.go [hunk 2]
```diff
@@ -424,7 +424,7 @@ sed -i s/PD_ADDR/${result}/g /data0/proxy.toml
 		script += fmt.Sprintf(str, pdAddr, tc.GetName(), tc.GetNamespace())
 	}
 
-	initContainers = append(initContainers, corev1.Container{
+	initializer := corev1.Container{
 		Name:  "init",
```
Comment (DanielZhangQD) [inline pkg/manager/member/tiflash_member_manager.go:428]: The name in L357 needs to change.

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 3]
```diff
@@ -579,6 +579,11 @@ type TiFlashSpec struct {
 	// +optional
 	Config *TiFlashConfigWraper `json:"config,omitempty"`
 
+	// Initializer is the configurations of the init container for TiFlash
+	//
+	// +optional
+	Initializer *InitContainerSpec `json:"initializer,omitempty"`
```
Comment (DanielZhangQD) [inline pkg/apis/pingcap/v1alpha1/types.go:585]:
```suggestion
	InitContainer *InitContainerSpec `json:"initContainer,omitempty"`
```

Comment (KanShiori) [reply pkg/apis/pingcap/v1alpha1/types.go:585]: `initContainers` is a field in `ComponentSpec`,so I think `initContainer` a little confusing

Comment (DanielZhangQD) [reply pkg/apis/pingcap/v1alpha1/types.go:585]: OK

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**